compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -543,68 +607,77 @@ LogicalResult insertLogicalObjectFifoAccess(ModuleOp moduleOp) {
         });
     for (AMDAIE::DmaCpyNdOp outputDmaOp : outputDmaOps) {
       Value sourceMemref = outputDmaOp.getSourceObjectFifo().getMemref();
-        memrefToLogicalObjectFifo[sourceMemref] =
-            std::make_pair(outputDmaOp.getSourceObjectFifo(),
-                           AMDAIE::MemoryAccess::Write);
+      memrefToLogicalObjectFifo[sourceMemref] = std::make_pair(
+          outputDmaOp.getSourceObjectFifo(), AMDAIE::MemoryAccess::Write);
     }
 
     // We maintain a map from AllocOp to LogicalObjectFifoAccessOp in order to
     // avoid creating a new LogicalObjectFifoAccessOp for the same AllocOp being
     // used in a different op.
     DenseMap<Value, AMDAIE::LogicalObjectFifoAccessOp>
         memrefToLogicalObjectFifoAccess;
+
     WalkResult res = coreOp->walk([&](Operation *op) {
-      if (isa<linalg::LinalgOp, vector::TransferReadOp, vector::TransferWriteOp,
-              memref::ExtractStridedMetadataOp>(op)) {
-        // We want to insert amdaie.logicalobjectfifo.access ops right before
-        // the first usage. But for vectorized ops this would mean they'd get
-        // inserted within the vectorized scf.for ops. We therefore would want
-        // to traverse to the outermost scf.for op in that case. Currently we
-        // bubble up this traversal till that operation whose parent is not a
-        // scf.for op. TODO: Generalize this later.
-        Operation *opToInsertRewriterPoint = op;
-        while (isa<scf::ForOp>(opToInsertRewriterPoint->getParentOp())) {
-          opToInsertRewriterPoint = opToInsertRewriterPoint->getParentOp();
+      bool hasAllocOperand = [op]() {
+        for (auto operand : op->getOperands()) {
+          auto definingOp = operand.getDefiningOp();
```

Comment:
I'd prefer the following :-
```suggestion
        for (Value operand : op->getOperands()) {
          Operation* definingOp = operand.getDefiningOp();
```